### PR TITLE
[BugFix] Remove broker load unnecessary cast

### LINKED ISSUE #9315: Broker load fail when set array column
### Steps to reproduce the behavior (Required)
1. 
``` sql
CREATE TABLE `tarray` (
  `k` int(11) NULL COMMENT "",
  `v1` ARRAY<varchar(65533)> NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`k`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`k`) BUCKETS 1
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false"
);
```
2. 
``` sql
LOAD LABEL t.xxx
(
    DATA INFILE('hdfs://x.x.x.x:9000/user/hive/warehouse/t/*')
    INTO TABLE tarray
    FORMAT AS "parquet"
    (k,v)
    set(v1=v)
)
WITH BROKER 'broker1'
PROPERTIES
(
    'timeout' = '3600'
)
```

### Expected behavior (Required)
load success

### Real behavior (Required)
load fail

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9315

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/Load.java|12|12|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/load/Load.java [hunk 1]
```diff
@@ -515,12 +515,30 @@ public static List<Column> getPartialUpateColumns(Table tbl, List<ImportColumnDe
...
     /**
      * @param excludedColumns: columns that the type should not be inferred from expr.
-     *                         Such as, the type of column from path is VARCHAR, whether it is in expr args or not,
+     *                         Such as, the type of column from path is VARCHAR,
+     *                         whether it is in expr args or not,
      *                         and column exists in both schema and expr args.
      */
-    private static void replaceSrcSlotDescType(Map<String, Expr> exprsByName, TupleDescriptor srcTupleDesc,
-                                               Set<String> excludedColumns) throws UserException {
+    private static void replaceSrcSlotDescType(Table tbl, Map<String, Expr> exprsByName, TupleDescriptor srcTupleDesc,
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/load/Load.java:522]: You can run the daily test to check the cast

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 13 / 13 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/Load.java|13|13|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 10 / 13 (76.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/Load.java|10|13|76.92%|[536, 537, 538]|

File: fe/fe-core/src/main/java/com/starrocks/load/Load.java (hunk 1 above)
Comment (meegoo) [reply fe/fe-core/src/main/java/com/starrocks/load/Load.java:522]: pass the pr test

**Review by chaoyli — DISMISSED** [review]

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
@mergify rebase

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 10 / 13 (76.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/Load.java|10|13|76.92%|[536, 537, 538]|

**Comment by meegoo** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by meegoo** [comment]
@mergify backport branch-2.3

**Pull request merged**